commodore/config.py

```diff
@@ -176,3 +179,16 @@ def print_deprecation_notices(self):
             for notice in self._deprecation_notices:
                 notice = tw.fill(notice)
                 click.secho(notice)
+
+    def register_component_deprecations(self, cluster_parameters):
+        for cname in self._component_aliases.values():
+            ckey = component_parameters_key(cname)
+            cmeta = cluster_parameters[ckey].get("_metadata", {})
+
+            if cmeta.get("deprecated", False):
+                msg = f"Component {cname} is deprecated."
+                if "replaced_by" in cmeta:
+                    msg += f" Use component {cmeta['replaced_by']} instead."
+                if "deprecation_guide" in cmeta:
```

Comment:
Maybe rename to parameter to `migration_guide`?

Reply:
I had it like that previously, but it seemed more ambiguous, migration guide could be used to point to migration guides for major component upgrades too.

Reply:
True.
It occurred to me that maybe there's not always a replacement available, so it may not always be a guide.
Maybe `deprecation_announcement` or `deprecation_notice_link` something along this way ?

Reply:
Hm, or just a free-form `deprecation_notice` which gets printed verbatim?